src/lib/data/projects.ts

```diff
@@ -228,3 +238,216 @@ export const getConditionFieldsForProjects = async <
 
   return result;
 };
+
+export interface ProjectBudgetSegmentBreakdown {
+  organization: OrganizationId;
+  globalCluster: GlobalClusterId;
+  governingEntity: GoverningEntityId;
+  amountUSD: number;
+}
+
+export const getProjectBudgetsByOrgAndCluster = async <
+  Data extends {
+    projectVersion: Pick<
+      ProjectData['projectVersion'],
+      'id' | 'currentRequestedFunds'
+    >;
+  }
+>({
+  database,
+  projects,
+  log,
+}: {
+  database: Database;
+  projects: Map<ProjectId, Data>;
+  log: SharedLogContext;
+}): Promise<Map<ProjectId, Array<ProjectBudgetSegmentBreakdown>>> => {
+  const projectVersionIds = [...projects.values()].map(
+    (p) => p.projectVersion.id
+  );
+
+  const segments = await database.budgetSegment.find({
+    where: {
+      projectVersionId: {
+        [Op.IN]: projectVersionIds,
+      },
+      name: BUDGET_SEGMENTATION_BY_ORG,
+    },
+  });
+
+  const segmentsByProjectVersion = groupObjectsByProperty(
+    segments,
+    'projectVersionId'
+  );
+
+  const breakdowns = await database.budgetSegmentBreakdown.find({
+    where: {
+      budgetSegmentId: {
+        [Op.IN]: segments.map((s) => s.id),
+      },
+    },
+  });
+
+  const breakdownsBySegment = groupObjectsByProperty(
+    breakdowns,
+    'budgetSegmentId'
+  );
+
+  const entities = await database.budgetSegmentBreakdownEntity.find({
+    where: {
+      budgetSegmentBreakdownId: {
+        [Op.IN]: breakdowns.map((b) => b.id),
+      },
+    },
+  });
+
+  const entitiesByBreakdown = groupObjectsByProperty(
+    entities,
+    'budgetSegmentBreakdownId'
+  );
+
+  const result = new Map<ProjectId, Array<ProjectBudgetSegmentBreakdown>>();
+
+  const pOrgs = await database.projectVersionOrganization.find({
+    where: {
+      projectVersionId: {
+        [Op.IN]: projectVersionIds,
+      },
+    },
+  });
+
+  const orgsByProjectVersion = groupObjectsByProperty(
+    pOrgs,
+    'projectVersionId'
+  );
+
+  for (const [projectId, p] of projects.entries()) {
+    // There should be only 1 segment with name BUDGET_SEGMENTATION_BY_ORG
+    // for any project (except those with no budget)
+    const segments = segmentsByProjectVersion.get(p.projectVersion.id);
+
+    // Handle $0 projects with no budget segments
+    if (!segments && p.projectVersion.currentRequestedFunds === '0') {
```

Comment:
Too bad we aren't ready to use `BigInt` here and also below where you do `if (sum.toString() !== p.projectVersion.currentRequestedFunds)`. We cannot even convert `p.projectVersion.currentRequestedFunds` into `BigInt`, because we cannot compare to `0n` literal, since our compilation target is lower than ES2020.